\documentclass[aspectratio=169,  notheorems, sOuRcEs]{RUCPresentation}

%\usepackage{pgfforeach}
% \usepackage{foreach}

\newcounter{chapter}


\usepackage{tikz}
\usetikzlibrary{calc}
\usetikzlibrary{angles}
\usetikzlibrary{positioning}

\usetikzlibrary{arrows.meta}
\usetikzlibrary{shapes.geometric}
\usetikzlibrary{overlay-beamer-styles}

\ProvideDocumentCommand\pgfmathsetnewmacro{ m m }{ \newcommand*{#1}{} \pgfmathsetmacro{#1}{#2} }
\providecommand*{\pgfmathsetnewlengthmacro}[2]{ \newcommand*{#1}{} \pgfmathsetlengthmacro{#1}{#2} }

\tikzset{>=Stealth} % for good looking LaTeX arrow head
% \tikzset{overlay, remember picture}


\usepackage{myMathQA}
\usepackage{graphicx}
\usepackage{siunitx}
\usepackage{subcaption}

\usepackage{hyperref}

% \usepackage[style=authoryear]{biblatex}
\usepackage{biblatex}
\addbibresource{mscbib.bib}

\DeclareSourcemap{
    \maps[datatype=bibtex]{
        \map[overwrite]{
            \step[fieldset=note, null]
        }
        \map[overwrite]{
            \step[fieldset=ppn_gvk, null]
        }
        \map[overwrite]{
            \step[fieldset=file, null]
        }
        \map[overwrite]{
            \step[fieldset=doi, null]
        }
        \map[overwrite]{
            \step[fieldset=issn, null]
        }
        \map[overwrite]{
            \step[fieldset=isbn, null]
        }
        \map[overwrite]{
            \step[fieldset=pagetotal, null]
        }
    }
}

% \DeclareCiteCommand{\beamerfootcite}
%   {}
%   {\footnote[frame]{\printbibliography[keys=\thefield{entrykey}]}}
%   {\multicitedelim}
%   {\usebibmacro{postnote}}

\ProvideDocumentCommand{\beamerfootcite}{ m }{\footnote[frame]{\ \fullcite{#1}}}
\renewcommand*{\nameyeardelim}{\addcomma\space} % cleaner authoryear style

% \usepackage{animate}



\def\k{\raisebox{1ex}{\rotatebox{180}{k}}}
% \newsavebox\tempboxbeamer

\usetheme{RUCTHEME}
% \usetheme{Warsaw}

\makeatletter
\renewcommand\@makefnmark{\hbox{\@textsuperscript{\usebeamercolor[fg]{footnote mark}\usebeamerfont*{footnote mark}[\@thefnmark]}}}
% \renewcommand\@makefntext[1]{{\usebeamercolor[fg]{footnote mark}\usebeamerfont*{footnote mark}[\@thefnmark]}\usebeamerfont*{footnote} #1}
\makeatother

\hypersetup{colorlinks = false, citecolor=TextColor}
\AtBeginBibliography{\small\color{TextColor}}


\def\dumbmatrixtwobytwo[#1,#2]{%
    \count6=#1\relax \count7=#2\relax%
    \count2=#1\relax \count3=#2\relax \count4=#1\relax \count5=#2\relax%
    \multiply\count2 by \count6\relax \multiply\count4 by \count7\relax%
    \multiply\count3 by \count6\relax \multiply\count5 by \count7\relax%
    \ensuremath{%
        \left[%
            \begin{matrix}%
                \number\count2 & \number\count4 \\%
                \number\count3 & \number\count5%
            \end{matrix}%
        \right]%
    }
}


\def\dumbmatrixthreebythree[#1,#2,#3]{%
    \count10=#1\relax \count11=#2\relax \count12=#3\relax
    \count1=#1\relax \count4=#1\relax \count7=#1\relax%
    \count2=#2\relax \count5=#2\relax \count8=#2\relax%
    \count3=#3\relax \count6=#3\relax \count9=#3\relax%
    \multiply\count1\count10\relax \multiply\count4\count11\relax \multiply\count7\count12\relax%%
    \multiply\count2\count10\relax \multiply\count5\count11\relax \multiply\count8\count12\relax%%
    \multiply\count3\count10\relax \multiply\count6\count11\relax \multiply\count9\count12\relax%%
    \ensuremath{%
        \left[%
            \begin{matrix}%
                \number\count1 & \number\count4 & \number\count7 \\%
                \number\count2 & \number\count5 & \number\count8 \\%
                \number\count3 & \number\count6 & \number\count9%
            \end{matrix}%
        \right]%
    }
}

\usepackage{multirow}
\newcounter{reaction}
\ProvideDocumentCommand\reactionterm{ o m m m m }{% --> Function for reaction lines
    \stepcounter{reaction} %
    \IfNoValueTF{#1}
    { \arabic{reaction} & #3 & #4 & \(t^+_{\arabic{reaction}}\of{\vec{x}}\) & #2 & \(\br{#5}\) \\ \hline }
    { \multirow{2}*{\arabic{reaction}} & #3 & #4 & \(t^+_{\arabic{reaction}}\of{\vec{x}}\) & #2 &  \multirow{2}*{\(\br{#5}\)} \\
                                       & #4 & #3 & \(t^-_{\arabic{reaction}}\of{\vec{x}}\) & #1 &                             \\ \hline }
}

\ProvideDocumentEnvironment{reactiontable}{}%
{ \begin{center} \begin{tabular}{c|r@{\ \(\to\)\ }lr@{\ \(=\)\ }lc} %
    \(I\) & \multicolumn{2}{c}{`Reaction'} %
          & \multicolumn{2}{c}{`Rate Law'} %
          & \(\vec{r}_I^{\mathrm{T}}\) \\\hline }%
{ \end{tabular} \end{center} \setcounter{reaction}{0} }


% Defining a new coordinate system for the page:
%
% --------------------------
% |(-1,1)    (0,1)    (1,1)|
% |                        |
% |(-1,0)    (0,0)    (1,0)|
% |                        |
% |(-1,-1)   (0,-1)  (1,-1)|
% --------------------------

% For beamer title
\newcommand\makebeamertitle{\frame{\maketitle}}%

% For sections?
\graphicspath{{./images/}{../Thesis/images/}{../Thesis/images/phase/}{../Thesis/images/ode/}{../Thesis/images/fpe/}}


\def\cfp#1{c_#1^*}


\title{%
    An exploration on the differences of stochastic and
    deterministic models
}
% \subtitle{  Mathematical\ \ Bioscience a MSc Thesis }
\subtitle{  My MSc Thesis Summerized }
% \newline supervisor: Morten Anderson}

\author{by Anakin}

%\mode<presentation>


\begin{document}


\makebeamertitle


\section{Introduction}

\begin{frame}{Who am I?}

    \begin{columns}[T]

        \column{0.5\textwidth}

        \begin{itemize}[<+->]
            \item Hi, I'm Anakin
                \begin{itemize}[<.->]
                    \item You've probably met me
                \end{itemize}
            \item Started at RUC during covid times
                \begin{itemize}[<.->]
                    \item Applied to RUC in 2020
                    \item For the MSc program `Molecular Genetics'
                \end{itemize}
            \item Masters reform happened
                \begin{itemize}[<.->]
                    \item Hard pivoted into math
                    \item Never particularly interested in it before
                    \item Genetics/Developmental biology is still my first love
                \end{itemize}
        \end{itemize}

        \column<4->{0.5\textwidth}

        What led to me actually thinking `math is cool'
        \begin{itemize}
            \item Michael Levin
                \begin{itemize}
                    \item Bioelectricty as determining force
                    \item `Xenobots' showing life is eldritch
                \end{itemize}
            \item C. H. Waddington
                \begin{itemize}
                    \item laid foundations for epigenetics and systems biology
                    \item Coined the 'epigenetic landscape' similar to potential
                        energy landscape for gene expression
                \end{itemize}
        \end{itemize}


    \end{columns}

\end{frame}

\begin{frame}{The Research Question}

    \begin{itemize}
        \item So why am \emph{I} interested in this
    \end{itemize}
    \pause
    \begin{enumerate}[<+->]
        \item It's a tangible problem
        \item I wanted to find an excuse to go `balls deep' into stochasticity
        \item I believe there is an inherent relationship between
            statistical manifolds and genetic phenotypes
    \end{enumerate}

    \pause
    \begin{block}{As a consequence, my research question}
        How can stochastic gene copy number dynamics underlying bacterial
        heteroresistance be accurately captured using a Fokker-Planck formulation,
        and what insights does this stochastic framework provide beyond deterministic ODE models.
    \end{block}

\end{frame}

% Mechanisms of heteroresistance:
%     Caused by frequent mutations other than spontaneous gene amplifications:
%         Point mutations
%         Insertion sequence insertions
%         Smal   Caused by frequent mutations other than spontaneous gene amplifications:
%         Point mutations
%         Insertion sequence insertions
%         Smalll deletions
%     Resistance mutations with low or no fitness cost:
%         Stable even in the absence of antibiotics
%         Leads to stable heteroresistance
%     High fitness cost associated with resistance mutation:
%         Resistant phenotype can revert in the absence of antibiotics
%         Second-site compensatory mutations can increase fitness
%         Concomitant loss of resistance
%         Generates a mixed population:
%             Rare resistant cells with reduced fitness (contain a resistance mutation)
%             Main population of susceptible cells with higher fitness (contain a resistance mutation and a compensatory mutation)
%         Revertants often have different genotypes and/or phenotypes from the original isolate
%     Unstable resistance:
%         Caused by intrinsic instability of the resistance mutation itself:
%             Spontaneous unstable gene amplifications
%             Causes unstable heteroresistance due to the cost and high instability of gene amplifications
%         Revertants have the same genotype and phenotype as the original isolate


\section{Motivation}

\subsection{Heteroresistence}

\begin{frame}{Heteroresistence}

    % Text\beamerfootcite{Levin2024} supposed to be citation

    % The presence of resistance genes (mutated or acquired genes) in bacterial
    % cells typically leads to measurable antibiotic resistance, creating a
    % generally predictable relationship between a bacterium's genetic makeup
    % (genotype) and its observed antibiotic susceptibility (phenotype).
    % However, there are important exceptions.
    % Sometimes even genetically identical bacterial cells within the same
    % population can display phenotypic heterogeneity (variability) in their
    % responses to antibiotics, complicating the clear-cut classification into
    % susceptible or resistant categories. Heteroresistance is a notable example
    % of this variability, wherein a bacterial isolate predominantly susceptible
    % to an antibiotic includes smaller subpopulations of cells with reduced
    % antibiotic susceptibility.


    \begin{itemize}[<+->]
        \item Presence of resistance genes in bacterial cells leads
            to measurable antibiotic resistance
            \begin{itemize}[<.->]
                \item Generally predictable relationship between bacterial
                    genotype and antibiotic resistance
                \item Exceptions exist
            \end{itemize}
        \item Genetically identical bacterial cells can display phenotypic
            heterogeneity in responses to antibiotics.
            % \begin{itemize}
            %     \item Complicates classification into susceptible or
            %         resistant categories
            % \end{itemize}
        \item Heteroresistance: a bacterial isolate predominantly
            susceptible to an antibiotic includes smaller subpopulations
            of cells with reduced antibiotic susceptibility\beamerfootcite{Andersson2019}
    \end{itemize}

\end{frame}

\begin{frame}

    What does heteroresistance look like?
    \begin{columns}[T]
        \column{0.5\framewidth}

        \begin{itemize}
            \item Resistance mutations with low or no fitness cost
                \begin{itemize}
                    \item Stable even in the absence of antibiotics
                    \item Leads to stable heteroresistance
                \end{itemize}
        \end{itemize}

        \column{0.5\framewidth}

        \begin{itemize}
            \item High fitness cost associated with resistance mutation
                \begin{itemize}
                    \item Rare resistant cells with reduced fitness
                        (contain a resistance mutation)
                    \item Main population of susceptible cells with higher
                        fitness (contain a resistance mutation and a
                        compensatory mutation)
                \end{itemize}
        \end{itemize}

    \end{columns}

\end{frame}

\subsection{Copy Numbers}

\begin{frame}{Gene Copy Numbers}

    \centering

    \vspace{1em}
    One of the most interesting methods is through the
    gene copy number GCN\beamerfootcite{Nicoloff2024}\par
    \includegraphics[width=0.7\textwidth]{cgn.png}\par
    So we literally see the copy count of the gene increasing


\end{frame}


\section{What do we know}


\begin{frame}{The approach}

    \begin{itemize}
        \item See if it can be modeled with ODEs
        \item See if it can be modeled with SDEs
        \item Compare
    \end{itemize}


\end{frame}


\begin{frame}{ODE system}

    \tikz[overlay] \node<1-> at (frame cs:-0.65,-1.35) {
        \begin{tikzpicture}[ scale = 1,
            draw = black!80,
            thick,
            shorten >=2pt,
            overlay
            ]

            \node[draw=black, circle] (c1) at (-1, 0) {\(c_1\)};
            \node[draw=black, circle] (c2) at ( 1, 0) {\(c_2\)};

            \node[draw=black] (m) at ( -1, -1.20) {\(m\)};
            \node[] (source) at ( -2, -1.20) {\(m_0\)};

            \draw[->] (c2) [bend left]to node[midway, below] {\(w_2\)} (c1);
            \draw[->] (c1) [bend left]to node[midway, above] {\(w_1\)} (c2);

            \draw[->] (c1)  [loop left]to node[midway, left]  {\(k_1\)} (c1);
            \draw[->] (c2) [loop right]to node[midway, right] {\(k_2\)} (c2);

            \draw[-|] (m) to node[pos=0.35] (mm) {} (c1);
            \draw[-|, rounded corners] (c2) |- node[midway,anchor=south west] {\(q_m\)} (m);
            % \draw[-|] (c2) to[out=270, in=0, looseness=2] node[midway,anchor=north west] {\(q_m\)} (m);

            \node[anchor=east] at (mm) {\(q_1\)};

            \draw[->] (source) -- (m);

    \end{tikzpicture}};


    \begin{columns}
        \column<2->{0.5\framewidth}
        \vspace{8em}\par

        \begin{align*}
            \ode{c_1} &= k_1 c_1\left({n_1 - c_t \over n_1}\right) - w_1 c_1 + w_2 c_2 - q_1 m c_1 \nonumber \\
            \ode{c_2} &= k_2 c_2\left({n_2 - c_t \over n_2}\right) - w_2 c_2 + w_1 c_1\nonumber\\
            \ode{m}   &= m_0 - q_m m c_2 \nonumber
        \end{align*}

        \column{0.5\framewidth}

    \end{columns}



    \begin{tikzpicture}[overlay, remember picture]

        \node<3-> at (frame cs:0.5, 0.34) {
                \includegraphics[width=0.4\textwidth]
                {ode:jesper:I6:C1:T17481880809603815.pdf}
            };
        \node<3-> at (frame cs:0.5, -0.40) {
                \includegraphics[width=0.4\textwidth]
                {ode:jesper:I32:C1:T17481997939903586.pdf}
            };

    \end{tikzpicture}

\end{frame}

\begin{frame}{Analytical solution}


    \begin{columns}[T]
        \column{0.5\framewidth}
        \begin{itemize}
            \item Skipping the details
            \item We recognize that the system likely has an asymptotic stability
            \item Nullclines are given by
                \begin{align*}
                    \ode* c_1 = 0 &\iff c_2 = {q_1 m c_1- \pa{\omega_1 - \k_1c_1}c_1 \over w_2 - \k_1 c_1}
                    \\
                    \ode* c_2 = 0 &\iff c_1 = -{\pa{\omega_{2} - \k_{2} c_2} c_2 \over w_1 - \k_{2}c_2}
                    \\
                    \ode* m = 0 &\iff m = {m_0 \over q_m c_2}
                \end{align*}
        \end{itemize}


        \column{0.5\framewidth}

    \end{columns}

    \begin{tikzpicture}[overlay, remember picture]

        \node<1> at (frame cs:0.5, 0.0) {
                \includegraphics[width=0.5\textwidth]
                {phase:jesper:I5:Cm0=0_k1=1.0_k2=2.0_n1=100.0_n2=100.0_w1=0.015_w2=0.15_q1=0.85_q2=0.85_:T0.pdf}
            };

        % \node<1> at (frame cs:0.5, -0.40) {
        %         \includegraphics[width=0.4\textwidth]
        %         {ode:jesper:I32:C1:T17481997939903586.pdf}
        %     };


    \end{tikzpicture}

\end{frame}

\begin{frame}{Analytic Cont.}


    \begin{columns}[T]
        \column{0.6\framewidth}
        \begin{itemize}
            \item I wasn't \alt<1>{smart}{\cancel{smart} patient} enough to
                solve it purely algebraically
                \begin{itemize}
                    \item Things kept canceling to themselves
                \end{itemize}
            \item<2-> Instead we define a level set in the phase space
                \begin{equation*}
                    \clm C_s:= \left\{(c_1, c_2, m) \,\big|\,
                    \ode* \br{c_1+c_2+m} = 0\right\}
                \end{equation*}
            \item<3-> Means, when \(m=m_0\pa{q_m c_2}^{-1}\) that all solutions
                belong to the embedding
                \begin{equation*}
                    {q_1 \over q_2} {c_1 \over c_2} m_0 =
                    \pa{k_1 c_1 + k_2 c_2}
                    - \pa{ \k_2  c_2 + \k_1  c_1 }  \pa{c_1 + c_2}
                \end{equation*}

        \end{itemize}


        \column{0.5\framewidth}

    \end{columns}

    \begin{tikzpicture}[overlay, remember picture]

        \node<2> at (frame cs:0.5, 0.0) {
                \includegraphics[width=0.5\textwidth]
                {phase:jesper:I2:Cm0=0_k1=1.0_k2=1.0_n1=100.0_n2=90.0_w1=0.15_w2=0.15_q1=0.85_q2=0.85_:T0.pdf}
            };

        \node<3-> at (frame cs:0.5, 0.34) {
                \includegraphics[width=0.4\textwidth]
                {ode:jesper:I32:C1fp:T17481997939903586.pdf}
            };
        % \node<3-> at (frame cs:0.5, -0.40) {
        %         \includegraphics[width=0.4\textwidth]
        %         {ode:jesper:I6:C1fp:T17481880809603815.pdf}
        %     };
    \end{tikzpicture}
\end{frame}

\begin{frame}{Analytic Cont.}


    \begin{columns}[T]
        \column{0.5\framewidth}
        \begin{itemize}
            \item From the fact that we can then make a Lyapunov
                out of the logistic growths, we know the system is
                asymptotically stable\beamerfootcite{Meiss2017}
            \item Intuitively its just from adding logistic growths
        \end{itemize}


        \column{0.5\framewidth}
        \begin{itemize}
            \item In the case of a non-hyperbolic fixed point at
                \(\vec{p}=\pa{0,0,0}\)
                % \begin{itemize}
                %     \item Really when \(m=0\)
                % \end{itemize}
            \item We use the fact that the subspaces of an invariant set
                are themselves invariant\footnotemark[2]
                and show instability with respect to the embedding
                \begin{equation*}
                    \left[
                        \begin{matrix}
                            \omega_1& w_2 & 0 \\
                            w_1 & \omega_2 & 0 \\
                            0 & 0 & 0
                        \end{matrix}
                    \right]
                    \to
                    \bmatform[\omega_1, w_2; w_1, \omega_2]
                \end{equation*}
        \end{itemize}

    \end{columns}

    % \begin{tikzpicture}[overlay, remember picture]
    %
    %     \node<2> at (frame cs:0.5, 0.0) {
    %             \includegraphics[width=0.5\textwidth]
    %             {phase:jesper:I2:Cm0=0_k1=1.0_k2=1.0_n1=100.0_n2=90.0_w1=0.15_w2=0.15_q1=0.85_q2=0.85_:T0.pdf}
    %         };
    %
    %     \node<3-> at (frame cs:0.5, 0.34) {
    %             \includegraphics[width=0.4\textwidth]
    %             {ode:jesper:I32:C1fp:T17481997939903586.pdf}
    %         };
    %     % \node<3-> at (frame cs:0.5, -0.40) {
    %     %         \includegraphics[width=0.4\textwidth]
    %     %         {ode:jesper:I6:C1fp:T17481880809603815.pdf}
    %     %     };
    % \end{tikzpicture}
\end{frame}




\section{Stochasticity}

\begin{frame}{Stochasticity}

    \begin{itemize}
        \item Stochasticity is a touchy topic mathematically
            \begin{itemize}[<.->]
                \item Fundementally nothing mathematically can have a
                    non-determined output
                \item Cue the concept of a process
            \end{itemize}
        \item If we tie the outcome of a state, to the current state and time,
            \begin{equation*}
                W\of{x\of t, t \,|\, x\of{t-1}}
            \end{equation*}
            we can pretend that randomness isn't real, but still use it mathematically
    \begin{align}
        %\ode x = A\of{x,t} + B\of{x,t}
        \ode x = A\of{x\of t, t} \implies \dd{x} &= A\of{x,t} \dd{t}
        \shortintertext{Throw our randomness in there}
        \dd{x} &= A\of{x,t} \dd{t} + \sqrt{  B\of{x,t} } \ \dd{W_t}
    \end{align}
    % from allowing our ODE to be influenced by stochastic noise, we now
    % have a SDE.

     \item More formally known as the the \emph{Langevin equation},
    \end{itemize}

    % So what some \emph{lunatic} did, is glue a Wiener process to the end of our
    % differential.
    % With some multivariate function \(B\), acting on \(W_t\) as a scaling factor
    % for the variance.
    % By substitution of variables, we can redefine the sides of the derivative,

\end{frame}

\begin{frame}{The Master Equation}


    The dynamics of a probability distribution with the Markov property
    have a \emph{master equation}\beamerfootcite{Gardiner1987}
    \begin{equation} \label{eq.mastereq}
        \pde* P\of{x,t} =\int_{-\infty}^{\infty}
        \left[W_t\of{x|x'}P\of{x',t} - W_t\of{x'|x} P\of{x,t}\right]
        \dd{x'}
    \end{equation}
    \begin{itemize}
        \item  \(W_t\of{x|x'}\) is the rate of state transitions from \(x'\)
            \begin{itemize}
                \item Also known as a Weiner process
            \end{itemize}
        \item  \(P\of{x, t}\) is the probability of \(x\) at the given time
    \end{itemize}

\end{frame}

\begin{frame}{The Chemical Master Equation}


    \setcounter{footnote}{1}
    But in the non-continous form, we can make a \emph{chemical}
    master equation\beamerfootcite{Gardiner1987}
    \begin{equation} \label{eq.chemmastereq}
        \pde* P\of{x,t}
        =
        \sum_{j}^{}\br{
            a_j\of{\vec{n} - \vec{v}_j} P\of{\vec{n} - \vec{v}_j,t}
            - a_j\of{\vec{n}} P\of{\vec{n},t}
        }
    \end{equation}

    \begin{itemize}
        \item \(a_j\) represents a `propensity'
        \item \(\vec{v}_j\) a state change vector
        \item \(\vec{n}_j\) a state change vector
        \item Basically just a discrete version of the master equation
            \begin{itemize}
                \item That was a lie
            \end{itemize}
    \end{itemize}

\end{frame}

\begin{frame}{Fokker-Planck}

    \only<1>{
    \begin{itemize}
        \item Taking the chemical master equation, taylor expanding it to second order
            \begin{equation*}
                \pde[t]{P\of{\vec{x},t}} =
                \sum_{r=1}^{R} \left[
                    \sum_{i=1}^{N} n_{ri} \pde*[x_i] w_r\of{\vec{x}} P\of{\vec{x},t}
                    +
                    \sum_{i=1}^{N} \sum_{j=1}^{N} {n_{ri} n_{rj} \over 2}
                    \pde*[x_i, x_j] w_r\of{\vec{x}} P\of{\vec{x},t}
                \right]
            \end{equation*}
        \item By smoothing this guy out, we get the fokker-plank
        \item Turning systems like this,
            \begin{equation*}
                A_1 \mathop\rightleftharpoons^{k_1}_{k_2} A_2
            \end{equation*}


    \end{itemize}
}

    \only<2>{
    \begin{itemize}
        \item To a stationary distribution, like
    \end{itemize}
    }

    \begin{tikzpicture}[overlay, remember picture]
        \node<2> at (frame cs:0,-0.25) {
                \includegraphics[width=0.85\framewidth]
                {Screenshot from 2025-05-26 10-00-15.png}
            };
    \end{tikzpicture}

\end{frame}



\subsection{The Ordinary Differential System}

\begin{frame}{Chemicafying it}


    \begin{columns}[T]
        \column{0.3\textwidth}
        \begin{itemize}
            \item Recontextualizing the compartment model into kinetics
        \end{itemize}
        \begin{align*}
            c_1 &\xrightleftharpoons[w_2]{w_1} c_2 \\
            c_1 + N &\xrightleftharpoons[k_{-1}]{k_1} 2c_1\\
            c_2 + N &\xrightleftharpoons[k_{-2}]{k_2} 2c_2
        \end{align*}

        \column{0.7\textwidth}
        \begin{itemize}
            \item Letting  \(\vec{r}_1 = \br{-1,1}^{\mathrm{T}} \) be our transition vector,
            \item The first kinetic reaction \(c_1 \xrightleftharpoons[w_2]{w_1} c_2 \)
            becomes
        \end{itemize}
        \begin{align*}
                T\of{ \vec{c} \to \vec{c} + \vec{r}_1} &= w_1 {c_1!\over {c_1 - 1}!}{c_2!\over {c_2 - 0}!} \\
            &\leadsto t_1^+\of{\vec{c}} = w_1 c_1 \\
                T\of{ \vec{c} \to \vec{c} - \vec{r}_1} &= w_2 {c_1!\over {c_1 - 0}!}{c_2!\over {c_2 - 1}!} \\
            &\leadsto t_1^-\of{\vec{c}} = w_2 c_2
        \end{align*}

    \end{columns}

\end{frame}

\begin{frame}

    \begin{columns}[T]
        \column{0.5\textwidth}

        \begin{itemize}
            \item The same process for the rest of the terms leads to a table
        \end{itemize}

        \begin{reactiontable}
            \reactionterm[\(w_2 c_2\)]{\(w_1 c_1\)}{\(c_1\)}{\(c_2\)}{-1, 1}
            \reactionterm[\(k_{-1} c_1^2\)]{\(k_1 n c_1\)}{\(c_1+N\)}{\(2 c_1\)}{ 1, 0}
            \reactionterm[\(k_{-2} c_2^2\)]{\(k_2 n c_2\)}{\(c_2+N\)}{\(2 c_2\)}{ 0, 1}
        \end{reactiontable}

        \column{0.5\textwidth}

    \end{columns}


    \begin{tikzpicture}[overlay, remember picture]


        \node at (frame cs:0.55, 0.45) {
                \includegraphics[width=0.38\framewidth, page=1]
                {ode/ssa:ode-3-2:I19:C2fp:T1747486672.pdf}
            };

        \node at (frame cs:0.55,-0.35) {
                \includegraphics[width=0.38\framewidth, page=3]
                {ode/ssa:ode-3-2:I19:C2fp:T1747486672.pdf}
            };

    \end{tikzpicture}

\end{frame}

\begin{frame}{We add Smothering}

    \begin{columns}[T]
        \column{0.5\textwidth}

        \begin{itemize}
            \item With smothering
        \end{itemize}

        \alt<1>{
            \begin{reactiontable}
                \reactionterm[\(w_2 c_2\)]{\(w_1 c_1\)}{\(c_1\)}{\(c_2\)}{-1, 1}
                \reactionterm[\(k_{-1} c_1^2\)]{\(k_1 n c_1\)}{\(c_1+N\)}{\(2 c_1\)}{ 1, 0}
                \reactionterm[\(k_{-2} c_2^2\)]{\(k_2 n c_2\)}{\(c_2+N\)}{\(2 c_2\)}{ 0, 1}
            \end{reactiontable}
        }{
            \begin{reactiontable}
                \reactionterm[\(w_2 c_2\)]{\(w_1 c_1\)}{\(c_1\)}{\(c_2\)}{-1, 1}
                \reactionterm[\(k_{-1} c_1^2\)]{\(k_1 n c_1\)}{\(c_1+N\)}{\(2 c_1\)}{ 1, 0}
                \reactionterm[\(k_{-2} c_2^2\)]{\(k_2 n c_2\)}{\(c_2+N\)}{\(2 c_2\)}{ 0, 1}
                \reactionterm{\(k_{1}c_1c_2\)}{\(c_1+c_2\)}{\(c_1+N\)}{-1, 0}
                \reactionterm{\(k_{2}c_1c_2\)}{\(c_1+c_2\)}{\(c_2+N\)}{0, -1}
            \end{reactiontable}
        }

        \column{0.5\textwidth}

        \only<1,2>{
            \begin{itemize}
                \item We can think about smothering them
            \end{itemize}
            \begin{align*}
                c_1 + c_2 &\xrightarrow{k_1} c_1 \\
                c_1 + c_2 &\xrightarrow{k_2} c_2
            \end{align*}
        }

    \end{columns}


    \begin{tikzpicture}[overlay, remember picture]

        \node<3-> at (frame cs:0.55, 0.45) {
                \includegraphics[width=0.38\framewidth, page=1]
                {ode/ssa:ode-5-2:I19:C2fp:T17475890580172482.pdf}
            };

        \node<3-> at (frame cs:0.55,-0.35) {
                \includegraphics[width=0.38\framewidth, page=3]
                {ode/ssa:ode-5-2:I19:C2fp:T17475890580172482.pdf}
            };

    \end{tikzpicture}

\end{frame}

\begin{frame}{Weird, what about...}

    \begin{columns}[T]
        \column{0.5\textwidth}

        \begin{itemize}
            \item With nutrition?
        \end{itemize}

        \alt<1>{
            \begin{reactiontable}
                \reactionterm[\(w_2 c_2\)]{\(w_1 c_1\)}{\(c_1\)}{\(c_2\)}{-1, 1}
                \reactionterm[\(k_{-1} c_1^2\)]{\(k_1 n c_1\)}{\(c_1+N\)}{\(2 c_1\)}{ 1, 0}
                \reactionterm[\(k_{-2} c_2^2\)]{\(k_2 n c_2\)}{\(c_2+N\)}{\(2 c_2\)}{ 0, 1}
                \reactionterm{\(k_{1}c_1c_2\)}{\(c_1+c_2\)}{\(c_1+N\)}{-1, 0}
                \reactionterm{\(k_{2}c_1c_2\)}{\(c_1+c_2\)}{\(c_2+N\)}{0, -1}
            \end{reactiontable}
        }{
            \begin{reactiontable}
                \reactionterm[\(w_2 c_2\)]{\(w_1 c_1\)}{\(c_1\)}{\(c_2\)}{-1, 1, 0}
                \reactionterm[\(k_{1} c_1^2\)]{\(k_1 n c_1\)}{\(c_1+N\)}{\(2 c_1\)}{ 1, 0, -1}
                \reactionterm[\(k_{2} c_2^2\)]{\(k_2 n c_2\)}{\(c_2+N\)}{\(2 c_2\)}{ 0, 1, -1}
                \reactionterm{\(k_{1}c_1c_2\)}{\(c_1+c_2\)}{\(c_1+N\)}{-1, 0, 1}
                \reactionterm{\(k_{2}c_1c_2\)}{\(c_1+c_2\)}{\(c_2+N\)}{0, -1, 1}
            \end{reactiontable}
        }

        \column{0.5\textwidth}


    \end{columns}


    \begin{tikzpicture}[overlay, remember picture]

        \node<2-> at (frame cs:0.55, 0.45) {
                \includegraphics[width=0.38\framewidth, page=1]
                {ode/ssa:ode-5-3:I19:C2:T1747490399601534.pdf}
            };

        \node<2-> at (frame cs:0.55,-0.35) {
                \includegraphics[width=0.38\framewidth, page=3]
                {ode/ssa:ode-5-3:I19:C2:T1747490399601534.pdf}
            };

    \end{tikzpicture}


\end{frame}

\begin{frame}{Oh, well what about treatment, ignoring nutrients}

        \only<1>{
    \begin{columns}[T]
        \column{0.3\textwidth}

        \begin{itemize}
            \item Antibiotic terms
                \begin{align*}
                    \emptyset &\xrightarrow{m_0} M \\
                    M+c_2 &\xrightarrow{q_m} c_2 \\
                    B + c_1  &\xrightarrow{q_1} M
                \end{align*}
            \item Can't avoid using antibiotics as a variable if we want
                to match the ODE system
        \end{itemize}


        \column{0.7\textwidth}

        \begin{reactiontable}
            \reactionterm[\(w_2 c_2\)]{\(w_1 c_1\)}{\(c_1\)}{\(c_2\)}{-1, 1, 0}
            \reactionterm[\(k_{1} c_1^2\)]{\(k_1 n c_1\)}{\(c_1+N\)}{\(2 c_1\)}{ 1, 0, 0}
            \reactionterm[\(k_{2} c_2^2\)]{\(k_2 n c_2\)}{\(c_2+N\)}{\(2 c_2\)}{ 0, 1, 0}
            \reactionterm{\(k_{1}c_1c_2\)}{\(c_1+c_2\)}{\(c_1+N\)}{-1, 0, 0}
            \reactionterm{\(k_{2}c_1c_2\)}{\(c_1+c_2\)}{\(c_2+N\)}{0, -1, 0}
            \reactionterm{\(m_0\)}{\(\emptyset\)}{\(M\)}{0, 0, 1}
            \reactionterm{\(q_m m c_2\)}{\(c_2 + M\)}{\(c_2+\emptyset \)}{0, 0, -1}
            \reactionterm{\(q_1 m c_1\)}{\(c_1 + M\)}{\(M+N\)}{-1, 0, 0}
        \end{reactiontable}


    \end{columns}
    }

        \only<2>{
            \begin{columns}[T]
                \column{0.5\textwidth}

                \begin{itemize}
                    \item Drift term is the same as ODE,
                \begin{align*}
                    \vec{A}\of{\vec{c}}
                    &=
                    \bmatform[%
                    \pa{n - (c_1 + c_2) } k_1 c_1 - \pa{ w_1 c_1 - w_2 c_2  }
                    -q_1 m c_1;%
                    \pa{n - (c_2 + c_1) } k_2 c_2 + \pa{ w_1 c_1 - w_2 c_2  };%
                    \pa{m_0 - q_m m c_2}
                    ]
                \end{align*}

                \vspace{7em}
                \end{itemize}

            %     {\small
            %     \begin{align*}
            %         \mat{B}\of{\vec{c}} &=
            %          \bmatform[%
            %          \pa{n +  c_1 + c_2} k_1 c_1 +  w_1 c_1 + w_2 c_2 + q_1 m c_1,
            %          - \pa{ w_1 c_1 + w_2 c_2  }, 0;
            %          - \pa{ w_1 c_1 + w_2 c_2  },
            %          \pa{n +  c_1  + c_2} k_2 c_2 + w_1 c_1 + w_2 c_2, 0;
            %          0, 0 , m_0 + q_m m c_2
            %          ]
            %     \end{align*}
            % }

                \column{0.5\textwidth}


            \end{columns}
    }

    \begin{tikzpicture}[overlay, remember picture]

        \node<2-> at (frame cs:-0.50, -0.35) {
                \includegraphics[width=0.55\framewidth, page=1]
                {ode/ssa:ode-8-3:I34:C2:T17476639695953887.pdf}
            };

        \node<2-> at (frame cs:0.50,-0.35) {
                \includegraphics[width=0.55\framewidth, page=2]
                {ode/ssa:ode-8-3:I34:C2:T17476639695953887.pdf}
            };

    \end{tikzpicture}


\end{frame}


\subsection{My Proposed Model}

\begin{frame}{My Model}

    % \raggedbottom



    \begin{columns}[T]
        \column<1->{0.8\textwidth}
        \vspace{1ex}
        \begin{itemize}
            \item Transcription happens at a rate one product/gene/process
            \item So lets model what happens in a system where
                gene copies actually vary
        \end{itemize}


        \column{0.2\textwidth}

    \end{columns}\par

    \begin{columns}[T]
        \column<2>{0.3\textwidth}
        \vspace{1ex}
        \begin{itemize}
            \item The `reactions' then
                \begin{align*}
                    X + N &\xrightleftharpoons[k_{-1}]{k_1} 2 X \\
                    X + N &\xrightarrow{k_2} X + Y \\
                    X + B &\xrightarrow{k_3} B \\
                    Y + B &\xrightarrow{k_4} N \\
                    Y    &\xrightarrow{k_5} N
                \end{align*}
        \end{itemize}


        \column{0.7\textwidth}


    \end{columns}

    \tikz[overlay, remember picture] \node at (frame cs:0.35,0)
    {%
        \begin{tikzpicture}
            \pgfmathsetnewmacro\genelength{35}
            \pgfmathsetnewmacro\geneheight{2}

            \pgfmathsetnewmacro\translength{10}
            \pgfmathsetnewmacro\transheight{2}

            \pgfmathsetnewmacro\reactionthickness{1}
            \pgfmathsetnewmacro\dnawidth{4}

            \pgfmathsetnewmacro\shortendist{6}
            \pgfmathsetnewmacro\bendstrength{1.05} % √(1²+(0.2-0.75)²)

            \pgfmathsetnewmacro\geneshift{5}

            \providecommand*\genecolor{red!80!black}
            \providecommand*\transcolor{green!80!black}

            \useasboundingbox (-4,-1.25) rectangle (5.5,1.35);
            \coordinate (O) at (0,0);

            \node[circle] (west) at (-3.5, 0) {};
            \node[circle] (east) at ( 4.5, 0) {};

            \node[anchor=south east] at (west) {\(5'\)};
            \node[anchor=south west] at (east) {\(3'\)};

            \draw[line width = \dnawidth] (west.center) to (east.center);

            \draw[line width = \dnawidth] (east.north) to (east.south);
            \draw[line width = \dnawidth] (west.north) to (west.south);


            \coordinate (g1) at (-2,0); \coordinate (g2) at ( 0,0);
            \coordinate (g3) at ( 3,0);

            \coordinate (t1) at (-3,0); \coordinate (t2) at (-1,0);
            \coordinate (t3) at ( 1,0);
            \coordinate (t4) at ( 2,0); \coordinate (t5) at ( 4,0);

            \draw[line width = \dnawidth+1, BackgroundColor, dashed] (t3) -- (t4);
            \draw[line width = \dnawidth+1, BackgroundColor] ([xshift=0.37cm]t3) to ([xshift=-0.43cm]t4);

            \foreach \POINT in {1,2,3}{
                \node[rectangle, fill=\genecolor, draw=\genecolor, minimum width=\genelength, minimum height=\geneheight] at (g\POINT) {};
            }

            \foreach \POINT in {1,2,3,4,5}{
                \node[rectangle, fill=\transcolor, draw=\transcolor, minimum width=\translength, minimum height=\transheight] at (t\POINT) {};
            }

            \node[above, yshift=5] (x1) at (g1) {\(X\)};
            \node[above, yshift=5] (x2) at (g2) {\(X\)};
            \node[above, yshift=5] (x3) at (g3) {\(X\)};

            \node  (B) at ( 4.75, 0.85) {\(B\)};
            \node (B2) at ( 0.50, 0.85) {\(B\)};

            \node  (N) at ( 2.00,-0.75) {\(N\)};
            \node (N2) at (-2.00, 1.00) {\(N\)};
            \node (N3) at ( 2.75, 1.00) {\(N\)};

            \node  (y) at ( 4.00,-0.75) {\(Y\)};
            \node (y2) at ( 1.50, 0.30) {\(Y\)};
            \node (y3) at ( 0.75,-0.75) {\(Y\)};

            \begin{scope}[line width=\reactionthickness, shorten <=\shortendist, shorten >=\shortendist]
                \draw[->] ([yshift=\geneshift]g1)
                    to[bend left] node[midway, sloped] (k1path) {}
                    ([yshift=\geneshift]g2);
                \draw[->] ([yshift=-\geneshift]g2)
                    to[bend left] node[midway] (k-1path) {}
                    ([yshift=-\geneshift]g1);
            \end{scope}

            \node[above] at (k1path) {\(k_1\)};
            \node[below] at (k-1path) {\(k_{-1}\)};

            \draw[line width=\reactionthickness, shorten >=6pt]
                (N2) to[bend right] ([yshift=2.3]k1path.center);

            % \draw[->,line width=\reactionthickness, shorten <=5.5pt]
            %     ([yshift=-2.6]k-1path.center) to[bend right] (N3);



            \begin{scope}[line width = \reactionthickness]
                \draw[shorten >=5pt, shorten <=4pt ]
                (B.center) -- (x3.center)
                node[pos=0.5, sloped] (midbx) {\(\bigtimes\)};
                \draw[shorten >=5pt, shorten <=4pt ]
                (B2.center) -- (y2.center)
                node[pos=0.5, sloped] (midby) {\(\bigtimes\)};

                \draw[->] (midbx.center) to[in=0, out=105, looseness=1.5] (N3);
                \draw[->] (midbx.center) to[in=160, out=105, looseness=3] (B);

                \draw[->] (midby.center) to[in=180, out=71, looseness=1.5] (N3);

                % \draw[->] (midxb.center) -- ++(105:0.5);
                % \draw[->] (midxb.center) to[bend right] (N2.east);

                \draw[->] (N) .. controls ++( 45:\bendstrength) and ++(135:\bendstrength) .. (y)
                node[midway, below] {\(k_2\)};
                \draw[->] (y3) -- (N) node[midway, above] {\(k_5\)};
            \end{scope}

            \node[anchor=north west] at (midbx) {\(k_3\)};
            \node[anchor=west] at ([yshift=2,xshift=1]midby) {\(k_4\)};


        \end{tikzpicture}
    };


\end{frame}

\begin{frame}{Drift and Diffusion}

    \def\fourAone{ { \pa{k'_1  - k_{-1} x} x - k'_3 x } }
    \def\fourAtwo{ { k'_2 x - \pa{k'_4  + k_5 } y} }
    \def\fourBxx{ { k'_1 x + k_{-1} x^2 + k'_3 x } }
    \def\fourBxy{ { 0 } }
    \def\fourByy{ { k'_2 x  + k'_4 y + k_5 y} }
    \def\fourAvec{{ \bmatform[ \fourAone ; \fourAtwo ] } }
    \def\fourBmat{{ \bmatform[ \fourBxx , \fourBxy ; \fourBxy , \fourByy] }}


    \begin{columns}[T]
        \column{0.5\textwidth}

    \begin{itemize}
    \item \(k_1':= k_1 n\)
    \item \(k_{2}':= k_2 n\)
    \end{itemize}

        \column{0.5\textwidth}

    \begin{itemize}
    \item \(k_3':= k_3 b\)
    \item \(k_{4}':= k_4 b\)
    \end{itemize}

    \end{columns}

    \begin{align*}
        \vec{A}\of{\vec{x}}
        &=
        \begin{bmatrix} 1 \\ 0 \end{bmatrix} \pa{k'_1 x - k_{-1} x^2}
        + \begin{bmatrix} 0 \\ 1 \end{bmatrix} k'_2 x
        + \begin{bmatrix} -1 \\ 0 \end{bmatrix} k'_3 x
        + \begin{bmatrix} 0 \\ -1 \end{bmatrix} k'_4 y
        + \begin{bmatrix} 0 \\ -1 \end{bmatrix} k_5 y
        \\
        &= \fourAvec \\
        \mathbf{B}\of{\vec{x}}
        &=
        \dumbmatrixtwobytwo[ 1,  0] \pa{ k'_1 x + k_{-1} x^2 }
        + \dumbmatrixtwobytwo[ 0,  1] k'_2 x
        + \dumbmatrixtwobytwo[-1,  0] k'_3 x
        + \dumbmatrixtwobytwo[ 0, -1] k'_4 y
        + \dumbmatrixtwobytwo[ 0, -1] k_5 y
        \\
        &= \fourBmat
    \end{align*}

\end{frame}

\begin{frame}{Stability}

\begin{equation*}
    \vec{x}^*:=
    \left[ \begin{matrix} x^* \\ y^* \end{matrix} \right]
    =
    -
    \left[
        \begin{matrix}
            \displaystyle {k'_3 - k'_1 \over k_{-1} }  &
            \displaystyle { k'_2\pa{k'_3 - k'_1} \over k'_{-1} k'_4}
        \end{matrix}
    \right]^{\mathrm{T}}
\end{equation*}
\begin{align*}
    \ode* \left[\begin{matrix} \delta x \\ \delta y \end{matrix} \right]
    &=
    \pde*[\bgroup x^*\egroup] \vec{A}\of{\vec{x}^*} \delta x +
    \pde*[\bgroup y^*\egroup] \vec{A}\of{\vec{x}^*} \delta y \\
    &= \bmatform[ k'_3 - k'_1 ; k'_2 ] \delta x
    +  \bmatform[ 0; -\pa{k'_4 + k_5}] \delta y
    =
    \left[
        \begin{matrix}
            % - 2k'_{-1} x^* + k'_1  - k'_3  & 0\\
            k'_3  - k'_1  & 0\\
            k'_2 & -\pa{k'_4 + k_5}
        \end{matrix}
    \right]
    \left[\begin{matrix} \delta x \\ \delta y \end{matrix} \right]
\end{align*}

\begin{align*}
    k_3 b < k_1 n &\implies \Re\of{\lambda_{1}} < 0\land \Re\of{\lambda_2} < 0
                  &&\text{ which is stable} \\
    k_3 b = k_1 n &\implies \Re\of{\lambda_{1}} = 0\land \Re\of{\lambda_2} < 0
                  &&\text{ which is `spectrally' stable}\\
    k_3 b > k_1 n &\implies \Re\of{\lambda_{1}} > 0 \land \Re\of{\lambda_{2}} < 0
                  &&\text{ which is a saddle}
\end{align*}

\end{frame}


\begin{frame}{What does it looks like in action?}

        \begin{figure}
            \centering
            \begin{subfigure}{\textwidth/3}
                \includegraphics[width=1\textwidth, page=2]%
                {five_var/ssa:5-2:b>n:I4:C1:T17474950116682749.pdf}\vfil
                \includegraphics[width=1\textwidth, page=1]%
                {five_var/ssa:5-2:b>n:I4:C1:T17474950116682749.pdf}
            \end{subfigure}%
            \hfil
            \begin{subfigure}{\textwidth/3}
                \includegraphics[width=1\textwidth, page=2]%
                {five_var/ssa:5-2:b=n:I2:C1:T17474930541817677.pdf}\vfil
                \includegraphics[width=1\textwidth, page=1]%
                {five_var/ssa:5-2:b=n:I2:C1:T17474930541817677.pdf}
            \end{subfigure}%
            \hfil
            \begin{subfigure}{\textwidth/3}
                \includegraphics[width=1\textwidth, page=2]%
                {five_var/ssa:5-2:b<n:I0:C1:T17474930032197933.pdf}\vfil
                \includegraphics[width=1\textwidth, page=1]%
                {five_var/ssa:5-2:b<n:I0:C1:T17474930032197933.pdf}
            \end{subfigure}%
            % \caption{%
            %     Results from simulating the 2 variable system outlined in
            %     \Cref{tab.fivepar_twovar}.
            %     All parameters are simplified to be equal to \(1\).
            %     The difference in evolution/steady state come from varying \(b\) in relation to
            %     \(n\).
            %     When \(b>n\), there is no distribution nor evolution.
            %     When \(b=n\), there is an ephemeral stability near the origin.
            %     When \(b<n\), there is a stable point.
            % }\label{fig.5part2varmany}
        \end{figure}

\end{frame}

\begin{frame}{MORE}
    \begin{tikzpicture}[overlay, remember picture]

        % \node<1-> at (frame cs:0, -0.45) {
        %         \includegraphics[width=0.38\framewidth, page=1]
        %         {five_var/ssa:5-3:I16:C2:T17475031669379826.pdf}
        %     };
        %
        %
        % \node<1-> at (frame cs:0, 0.45) {
        %         \includegraphics[width=0.38\framewidth, page=1]
        %         {five_var/ssa:5-3:I20:C2:T17475033100262814.pdf}
        %     };

        \node<1-> at (frame cs:0.50,0.0) {
                \includegraphics[width=0.45\framewidth, page=3]
                {five_var/ssa:5-3:I16:C2:T17475031669379826.pdf}
            };

        \node<1-> at (frame cs:-0.50,0.) {
                \includegraphics[width=0.45\framewidth, page=3]
                {five_var/ssa:5-3:I20:C2:T17475033100262814.pdf}
            };

    \end{tikzpicture}
\end{frame}

\section{Discussion}

\begin{frame}{ODE to SDE Differences}


    \begin{itemize}
        \item From what we can see, a lot of the dynamics of the ode system
            were not preserved
        \item Hell, when the anti-biotic was added, it looks like we start
            seeing lokta-volterra esque dynamics
    \end{itemize}

    \vspace{8em}

    \begin{tikzpicture}[overlay, remember picture]

        \node<2-> at (frame cs:-0.50, -0.35) {
                \includegraphics[width=0.55\framewidth, page=1]
                {ode/ssa:ode-8-3:I34:C2fp:T17476639695953887.pdf}
            };

        \node<2-> at (frame cs:0.50,-0.35) {
                \includegraphics[width=0.55\framewidth, page=2]
                {ode/ssa:ode-8-3:I34:C2fp:T17476639695953887.pdf}
            };

    \end{tikzpicture}

\end{frame}

\begin{frame}{Limitations of my model}

    \begin{itemize}
        \item In the current form, what I can present,
            it is difficult to specifically find these rates
        \item A lot of assumptions were made
    \end{itemize}


\end{frame}




\section{End}
\begin{frame}{Questions?}
    \large Questions?
\end{frame}


\section{Bibliography}
\begin{frame}{Bibliography}

    \printbibliography

\end{frame}

\end{document}
